torch/export/__init__.py

```diff
@@ -59,6 +60,11 @@
 PassType = Callable[[torch.fx.GraphModule], Optional[PassResult]]
 
 
+class InputDim(NamedTuple):
```

Comment:
Can we replace it with the new _Dim object introduced in this PR?

https://github.com/pytorch/pytorch/pull/108448

Reply:
This would be a bigger change, I think. Right now `InputDim` represents something like `arg_name.shape[k]` but `_Dim` objects are more like symbols. 

Because internally symbolic shapes uses symbols, we should be able to bypass constraints altogether and represent information much more directly. Instead the current implementation of `dynamic_shapes` goes through constraints, which then get translated to symbols internally again.

Reply:
To make this change quicker, we have to remove constraints.